### PR TITLE
refactor: replace plate value with vehicleId value

### PR DESCRIPTION
**Pull request by b-gantz**
## Description

refactor functions in storage.lua to pass vehicleId instead of vehicle plate

## Checklist

- [ ] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

File: server/storage.lua [hunk 1]
```diff
@@ -54,10 +54,16 @@ function FetchVehicleEntityByPlate(plate)
     return MySQL.single.await('SELECT * FROM player_vehicles WHERE plate = ?', {plate})
 end
 
----@param plate string
+---@param vehicleId integer
+---@return VehicleEntity
+function FetchVehicleEntutyByVehicleId(vehicleId)
```
Comment (mafewtm) [inline server/storage.lua]: You spelt entity wrong.

**Comment by b-gantz** [comment]
@Manason according to our chat on DC, take a look

File: server/storage.lua (hunk 1 above)
Comment (b-gantz) [reply server/storage.lua]: Corrected in latest commit

File: server/storage.lua [hunk 2]
```diff
@@ -54,10 +54,16 @@ function FetchVehicleEntityByPlate(plate)
     return MySQL.single.await('SELECT * FROM player_vehicles WHERE plate = ?', {plate})
 end
 
----@param plate string
+---@param vehicleId integer
+---@return VehicleEntity
+function FetchVehicleEntityByVehicleId(vehicleId)
+    return MySQL.single.await('SELECT * FROM player_vehicles WHERE vehicleId = ?', {vehicleId})
+end
```
Comment (Manason) [inline server/storage.lua]: A new function is created but not used?

File: server/storage.lua [hunk 3]
```diff
@@ -54,10 +54,16 @@ function FetchVehicleEntityByPlate(plate)
...
+
+---@param vehicle integer|string
 ---@return boolean
-function DoesVehicleEntityExist(plate)
-    local count = MySQL.scalar.await('SELECT COUNT(*) FROM player_vehicles WHERE plate = ?', {plate})
+function DoesVehicleEntityExist(vehicle)
+    local count = MySQL.scalar.await('SELECT COUNT(*) FROM player_vehicles WHERE vehicleId = ? or plate = ?', {vehicle, vehicle})
```
Comment (Manason) [inline server/storage.lua:60]: What about the caller? Doesn't the parameter need to be vehicleId? Passing in the vehicle would be the entityId. vehicleId is a auto incremented column on the player_vehicles table and can be gotten using 

Entity(vehicle).state.vehicleid or exports.qbx_vehicles:GetVehicleIdByPlate(GetVehicleNumberPlateText(vehicle))

**Review by Manason — CHANGES_REQUESTED** [review]

File: server/storage.lua (hunk 2 above)
Comment (b-gantz) [reply server/storage.lua]: Yeah, as a future replacement for `FetchVehicleEntityByPlate` which I guess I should mark as `depracated` maybe?

File: server/storage.lua (hunk 3 above)
Comment (b-gantz) [reply server/storage.lua:60]: It is used in https://github.com/Qbox-project/qbx_vehicleshop/blob/f75a0090b87928406a1e86cb7f0a516fa093944a/server/main.lua#L85 which I wasn't going to edit (maybe I should) and there is plate, so I tried to handle both cases with one input variable

File: server/storage.lua (hunk 2 above)
Comment (Manason) [reply server/storage.lua]: I think this is a situation where we can just change the callers with the function as well. This isn't an API in which we need to worry about breaking compatibility.

File: server/storage.lua (hunk 3 above)
Comment (Manason) [reply server/storage.lua:60]: DoesVehicleEntityExist is used for generating a unique plate when buying a vehicle. However, https://docs.qbox.re/resources/qbx_vehicles/exports/server#createplayervehicle already guarantees a unique plate, so the calling function can be removed.

File: server/storage.lua (hunk 2 above)
Comment (b-gantz) [reply server/storage.lua]: Fixed in server/main and in here

File: server/main.lua [hunk 4]
```diff
@@ -465,7 +465,7 @@ lib.addCommand('transfervehicle', {help = locale('general.command_transfervehicl
 
     local player = exports.qbx_core:GetPlayer(src)
     local target = exports.qbx_core:GetPlayer(buyerId)
-    local row = FetchVehicleEntityByPlate(plate)
+    local row = FetchVehicleEntityByVehicleId(vehicleId)
```
Comment (Manason) [inline server/main.lua:468]: Where is vehicleId declared?

**Comment by Manason** [comment]
Obseleted by #87

**Pull request closed without merging**